62_CAD/DrawResourcesFiller.cpp

```diff
@@ -684,6 +787,33 @@ uint32_t DrawResourcesFiller::addLineStyle_Internal(const LineStyleInfo& lineSty
 	return currentLineStylesCount++;
 }
 
+uint32_t DrawResourcesFiller::addDTMSettings_Internal(const DTMSettingsInfo& dtmSettingsInfo, SIntendedSubmitInfo& intendedNextSubmit)
+{
+	DTMSettings dtmSettings;
+
+	// TODO: this needs to be redone.. what if submit happens after that line?
+	// we need to make sure somehow that function below will not submit, we need both outline and contour styles in GPU memory
```

Comment:
Yes you're right, even if you have many line styles and the first one fits into men, the second one might evict everything. 
In the case that remainingLineStyles is < 2, do the auto submit calls and resets. And then call addLineStyle_Internal for both (asset they succeed, because they definitely should)

Also be careful to not call this in the middle of adding geometry that should reference this dtm settings.